### PR TITLE
Battery Model - Thermal Affects

### PR DESCRIPTION
**Pull request by teubert**
Added thermal affects to the Battery Model Based on edits from Chetan. 

Note: Required edits form the MBP_model_access and SaveLastState branches (pull requests are open). Once those are approved and merged, rebase this on the updated dev, then this PR can be reviewed/approved

### CODE REVIEW

**Comment by teubert** [comment]
Rebased dev after merging of SaveLastState branch- so now this branch should be functional- it can be merged after MBP_model_access

Note: before merging 1) merge MBP_Model_access into dev, 2) Rebase off of dev, 3) change the target branch of this PR to dev, and 4) Update tests (Chetan is providing some numbers).

Tests failing is because we need to update tests (see 4 above)

**Review by jason-watkins — APPROVED** [review]

**Review by chetankul — APPROVED** [review]
The updated changes look good to me. My only comment is regarding the Tau=100. Is this parameter user defined.

**Pull request merged**
